### DISCUSSION TITLE
Windows installation?

### DISCUSSION THREAD

**Discussion by johny2nd** [aquasecurity/trivy#5210] [Q&A] [answered]
### Question

I installed on windows with go and following install from source on trivy installation page. However, when I run trivy version i get this as the result. 
I know I can run it in wsl which i have done. But would like to be able to do it in windows native. 
Version: dev
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-09-18 18:15:07.820427384 +0000 UTC
  NextUpdate: 2023-09-19 00:15:07.820426884 +0000 UTC
  DownloadedAt: 2023-09-18 20:44:51.9597333 +0000 UTC
Java DB:
  Version: 1
  UpdatedAt: 2023-09-18 00:53:57.389614804 +0000 UTC
  NextUpdate: 2023-09-21 00:53:57.389614304 +0000 UTC
  DownloadedAt: 2023-09-18 20:45:43.5791572 +0000 UTC
  

### Target

None

### Scanner

Vulnerability

### Output Format

None

### Mode

Standalone

### Operating System

windows 

### Version

```bash
Version: dev
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-09-18 18:15:07.820427384 +0000 UTC
  NextUpdate: 2023-09-19 00:15:07.820426884 +0000 UTC
  DownloadedAt: 2023-09-18 20:44:51.9597333 +0000 UTC
Java DB:
  Version: 1
  UpdatedAt: 2023-09-18 00:53:57.389614804 +0000 UTC
  NextUpdate: 2023-09-21 00:53:57.389614304 +0000 UTC
  DownloadedAt: 2023-09-18 20:45:43.5791572 +0000 UTC
```

**Comment by knqyf263** [comment]
>However, when I run trivy version i get this as the result.

What is the problem? It seems to work.

**Comment by johny2nd** [comment]
Shouldn’t the version say what version it is, not just dev?


Regards,John Wagner

On Tuesday, September 19, 2023, 2:29 AM, Teppei Fukuda ***@***.***> wrote:





However, when I run trivy version i get this as the result.


What is the problem? It seems to work.

**Reply by knqyf263** [reply]
If you compile it from the source, the version will be dev.

**Comment by nikpivkin** [comment]
@johny2nd We set the version when building Trivy by passing flags to the linker. But since you are installing from source, this does not happen. You can install Trivy from the Github [releases](https://aquasecurity.github.io/trivy/v0.45/getting-started/installation/#install-from-github-release-official).

**Comment by johny2nd** [comment]
Is there a more recommended way for installing on windows without using wsl? 


Regards,John Wagner

On Tuesday, September 19, 2023, 7:23 AM, Nikita Pivkin ***@***.***> wrote:




@johny2nd We set the version when building Trivy by passing flags to the linker. But since you are installing from source, this does not happen. You can install Trivy from the Github releases.

**Reply by nikpivkin** [reply, marked as answer]
Trivy has a [build](https://github.com/aquasecurity/trivy/releases/download/v0.45.1/trivy_0.45.1_windows-64bit.zip) for Windows.

**Comment by johny2nd** [comment]
Thank you for the zip, could you also send me the link to where that is located?Also, How would I do I execute it?
    On Tuesday, September 19, 2023 at 07:31:42 AM EDT, Nikita Pivkin ***@***.***> wrote:  
 
 


Trivy has a build for Windows.

**Reply by nikpivkin** [reply]
@johny2nd https://github.com/aquasecurity/trivy/releases/tag/v0.45.1. Unpack the archive.

**Comment by johny2nd** [comment]
additionally, how do i update/check for updates to newest versions on windows?
    On Tuesday, September 19, 2023 at 09:05:35 AM EDT, john wagner II ***@***.***> wrote:  
 
  Thank you for the zip, could you also send me the link to where that is located?Also, How would I do I execute it?
    On Tuesday, September 19, 2023 at 07:31:42 AM EDT, Nikita Pivkin ***@***.***> wrote:  
 
 


Trivy has a build for Windows.

**Comment by johny2nd** [comment]
So, for windows there isnt a way in command line to check for latest update and install ?

    On Tuesday, September 19, 2023 at 09:41:32 AM EDT, Nikita Pivkin ***@***.***> wrote:  
 
 


@johny2nd https://github.com/aquasecurity/trivy/releases/tag/v0.45.1. Unpack the archive.

**Reply by knqyf263** [reply]
Scoop or Chocolatey
https://repology.org/project/trivy/versions
